js/foundation.magellan.js

```diff
@@ -165,8 +165,10 @@ class Magellan extends Plugin {
     }
 
     this.$active.removeClass(this.options.activeClass);
-    this.$active = this.$links.filter('[href="#' + this.$targets.eq(curIdx).data('magellan-target') + '"]').addClass(this.options.activeClass);
-
+    this.$active = this.$links.filter('[href="#' + this.$targets.eq(curIdx).data('magellan-target') + '"]').parents()[this.options.activeParents]
+  	               ? $( this.$links.filter('[href="#' + this.$targets.eq(curIdx).data('magellan-target') + '"]').parents()[this.options.activeParents] ).addClass(this.options.activeClass)
+  	               : this.$links.filter('[href="#' + this.$targets.eq(curIdx).data('magellan-target') + '"]').addClass(this.options.activeClass);
+    
     if(this.options.deepLinking){
```

Comment:
You shouldn't travel up the complete DOM tree twice!
I would rather write it this way and use jQuery's `eq()` so you don't have to recreate a jQuery object.

```javascript
this.$active = this.options.activeParents
                ? this.$links.filter('[href="#' + this.$targets.eq(curIdx).data('magellan-target') + '"]').parents().eq(this.options.activeParents).addClass(this.options.activeClass)
                : this.$links.filter('[href="#' + this.$targets.eq(curIdx).data('magellan-target') + '"]').addClass(this.options.activeClass);
```